src/modules/keyboardmanager/common/LayoutMap.cpp

```diff
@@ -155,3 +160,79 @@ void LayoutMap::UpdateLayout()
     keyboardLayoutMap[0xFF] = L"Undefined";
     // To do: Add IME key names
 }
+
+// Function to return two vectors: the list of names of all the keys for the drop down, and their corresponding virtual key codes. If the first argument is true, then an additional None option is added at the top
+std::pair<Windows::Foundation::Collections::IVector<Windows::Foundation::IInspectable>, std::vector<DWORD>> LayoutMap::GetKeyList(const bool isShortcut)
+{

```

Comment:
I would change this to two separate functions each returning different vectors. Otherwise the work is done twice for no good reason. Also, I think it might be a good idea to return the results ordered alphabetically, to make searching the combo box easier.

Reply:
Also, you can save a bunch of code by using `std::transform` and just mapping the values from one vector to another, in any case

Reply:
> Also, I think it might be a good idea to return the results ordered alphabetically, to make searching the combo box easier.

@saahmedm suggested to order them in categories, since for the shortcuts you need to first choose a modifier, and then choose an action key. Having the None option at the top also makes it easy to select that to remove the drop down. Making it alphabetical has the issue that alot of keys which are rarely used like the Attn key (I've never even seen it) and some other keys like that would appear in the middle of the character keys.

Reply:
Currently kept it in the same order, but the order is now constrained to the order generated by the initial language when the UI drop downs are first accessed. The constraint was added because some languages like French have some keys which don;t have unicode rep, but it exists in language, so the ordering was changing and causing issues. I tried handling dynamic ordering but the selection cannot be updated in the DropDownOpened as it leads to unexpected behaviour.
Also the function in the comment was split into two, and the one for the name vector just iterates over the codes and does a `box_value[keyboardMap[i]]`. Transform wasn't used since an IVector had to be returned and not an std::vector.

I think we can change the ordering after we get this change in, since that logic is now separated out it is easier to change.